app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -419,52 +427,19 @@ def ensure_managers
     ensure_network_manager
     ensure_cinder_manager
     ensure_swift_manager
+    # TODO: remove when this moves to before_initialization
     ensure_managers_zone_and_provider_region
   end
 
-  def ensure_managers_zone_and_provider_region
-    if network_manager
-      network_manager.zone_id         = zone_id
-      network_manager.tenant_id       = tenant_id
-      network_manager.provider_region = provider_region
-    end
-
-    if cinder_manager
-      cinder_manager.zone_id         = zone_id
-      cinder_manager.tenant_id       = tenant_id
-      cinder_manager.provider_region = provider_region
-    end
-
-    if swift_manager
-      swift_manager.zone_id         = zone_id
-      swift_manager.tenant_id       = tenant_id
-      swift_manager.provider_region = provider_region
-    end
-  end
-
-  def ensure_network_manager
-    build_network_manager(:type => 'ManageIQ::Providers::Openstack::NetworkManager') unless network_manager
+  # sigh, methods like this always start out innocent enough
+  def child_manager_references
+    [network_manager, cinder_manager, swift_manager].compact
   end
 
-  def ensure_cinder_manager
-    return false if cinder_manager
-    build_cinder_manager(:type => 'ManageIQ::Providers::Openstack::StorageManager::CinderManager')
-    true
-  end
-
-  def ensure_swift_manager
-    return false if swift_manager
-    build_swift_manager(:type => 'ManageIQ::Providers::StorageManager::SwiftManager')
-    true
-  end
-
-  after_save :save_on_other_managers
-
-  def save_on_other_managers
-    storage_managers.update_all(:tenant_mapping_enabled => tenant_mapping_enabled)
-    if network_manager
-      network_manager.tenant_mapping_enabled = tenant_mapping_enabled
-      network_manager.save!
+  def ensure_managers_zone_and_provider_region
+    child_manager_references.each do |child_manager|
```

Comment:
If we could work out the list of child managers generically we could do this in core right?

Reply:
I had child_managers but that messed with an already defined child_managers

here, we want to access the actual child managers that we are using.
If we want to stick with the child_managers association, then we could define swift_manager to be a select into child_managers or something like that?